app/models/vm_migrate_workflow.rb

```diff
@@ -16,13 +16,16 @@ def self.default_dialog_file
   def get_source_and_targets(refresh = false)
     return @target_resource if @target_resource && refresh == false
 
-    ems = @values[:src_ids].to_miq_a.collect { |v_id| v = Vm.find_by_id(v_id); v.ext_management_system }.uniq.compact
+    vms = Vm.where(:id => @values[:src_ids].to_miq_a)
+
+    ems_ids = Vm.where(:id => @values[:src_ids]).distinct.pluck(:ems_id)
```

Comment:
Think this would just be: `ems_ids = vms.distinct.pluck(:ems_id)`

Or better yet could this be combined with the following line:
```ruby
emses = ExtManagementSystem.where(:id =>vms.pluck(:ems_id)).distinct.compact
```